### PR TITLE
CP-49362: Implement xenserver_vdi provider resource

### PR DESCRIPTION
**Pull request by xueqingz**
test result:
```
ubuntu@ubuntu-server:~/code/terraform-provider-xenserver$ make testacc
if [ -z "/home/ubuntu/go/bin" ]; then echo "GOBIN is not set" && exit 1; fi
go mod tidy
go install .
ls -l /home/ubuntu/go/bin/terraform-provider-xenserver
-rwxrwxr-x 1 ubuntu ubuntu 22554841 Jun 19 06:51 /home/ubuntu/go/bin/terraform-provider-xenserver
source .env && TF_ACC=1 go test ./xenserver/ -v   -timeout 120m
=== RUN   TestAccNetworkResource
--- PASS: TestAccNetworkResource (1.69s)
=== RUN   TestAccPifDataSource
--- PASS: TestAccPifDataSource (0.67s)
=== RUN   TestAccSRDataSource
--- PASS: TestAccSRDataSource (0.68s)
=== RUN   TestAccNFSResource
--- PASS: TestAccNFSResource (5.29s)
=== RUN   TestAccSRResourceLocal
--- PASS: TestAccSRResourceLocal (3.94s)
=== RUN   TestAccSRResourceShared
--- PASS: TestAccSRResourceShared (5.24s)
=== RUN   TestAccVDIResource
--- PASS: TestAccVDIResource (4.03s)
=== RUN   TestAccVMResource
--- PASS: TestAccVMResource (1.75s)
PASS
ok      terraform-provider-xenserver/xenserver  23.307s
```

### CODE REVIEW

**Review by acefei — APPROVED** [review]

File: examples/terraform-main/main.tf [hunk 1]
```diff
@@ -91,3 +91,31 @@ resource "xenserver_network" "network" {
...
+
+output "vdi_out1" {
+  value = xenserver_vdi.vdi1
+}
+
+resource "xenserver_vdi" "vdi2" {
+  name_label       = "Test VDI2"
+  name_description = "A test VDI on Local storage"
+  sr_uuid          = data.xenserver_sr.sr.data_items[0].uuid
+  virtual_size     = 1 * 1024 * 1024 * 1024
```
Comment (xihuan-citrix) [inline examples/terraform-main/main.tf:114]: Is it mean 1G? i.e. the unit is only Byte?

**Review by xihuan-citrix — APPROVED** [review]

File: examples/terraform-main/main.tf (hunk 1 above)
Comment (xihuan-citrix) [reply examples/terraform-main/main.tf:114]: Or we can accept e.g. virtual_size     = 500 MB?

Comment (xueqingz) [reply examples/terraform-main/main.tf:114]: yes, you can check in vdi.md:
` `virtual_size` (Number) The size of virtual disk image (in bytes)`

Comment (xueqingz) [reply examples/terraform-main/main.tf:114]: We can't accept  `virtual_size = 500 MB` .

**Pull request merged**